File: FrontOffice/src/app/services/reclamationservice.ts
// reclamation.service.ts

import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Reclamation, StatusReclamation } from '../entities/Reclamation';

@Injectable({
  providedIn: 'root',
})
export class ReclamationService {
  private apiUrl = 'http://localhost:8080/reclamations';

  constructor(private http: HttpClient) {}

  getAllReclamations(): Observable<Reclamation[]> {
    return this.http.get<Reclamation[]>(`${this.apiUrl}`);
  }

  getReclamationById(id: number): Observable<Reclamation> {
    return this.http.get<Reclamation>(`${this.apiUrl}/${id}`);
  }

  addReclamation(reclamation: Reclamation): Observable<Reclamation> {
    return this.http.post<Reclamation>(`${this.apiUrl}`, reclamation);
  }

  updateReclamation(id: number, reclamation: Reclamation): Observable<Reclamation> {
    return this.http.put<Reclamation>(`${this.apiUrl}/${id}`, reclamation);
  }

  deleteReclamation(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }

  // Additional methods if needed

  getProcessedReclamations(): Observable<Reclamation[]> {
    return this.http.get<Reclamation[]>(`${this.apiUrl}/processed`);
  }

  getPendingReclamations(): Observable<Reclamation[]> {
    return this.http.get<Reclamation[]>(`${this.apiUrl}/pending`);
  }

  archiveResolvedReclamations(): Observable<Reclamation[]> {
    return this.http.post<Reclamation[]>(`${this.apiUrl}/archive/resolved`, {});
  }

  followReclamationStatus(reclamationId: number): Observable<Reclamation> {
    return this.http.get<Reclamation>(`${this.apiUrl}/follow/${reclamationId}`);
  }
}

File: BackEnd/src/main/java/com/example/pifinance_back/Services/ReclamationService.java
package com.example.pifinance_back.Services;

import com.example.pifinance_back.Entities.Reclamation;
import com.example.pifinance_back.Entities.statusReclamation;

import java.util.List;

public interface ReclamationService {
    Reclamation addReclamation(Reclamation reclamation);

    List<Reclamation> getAllReclamations();

    void deleteReclamation(int reclamationId);

    Reclamation updateReclamation(Reclamation reclamation);

    Reclamation getReclamationById(int reclamationId);

    List<Reclamation> generateProcessedReclamationsReport();

    List<Reclamation> managePendingReclamations();

    List<Reclamation> archiveResolvedReclamations();

    Reclamation followReclamationStatus(int reclamationId);

    Reclamation updateReclamationStatus(int id, statusReclamation newStatus);
}
